src/Symfony/Component/Uid/Tests/UuidTest.php

```diff
@@ -94,6 +94,22 @@ public function testEquals()
         $this->assertFalse($uuid1->equals($uuid2));
     }
 
+    /**
+     * @dataProvider provideInvalidEqualType
+     */
+    public function testEqualsAgainstOtherType($other)
+    {
+        $this->assertFalse((new Uuid(self::A_UUID_V4))->equals($other));
```

Comment:
@hhamon what about sub classes? Should a different class with the same UUID value conceptually be considered equal?

Reply:
from the pov of the Uuid class: yes
from the pov of the child class: theirs to decide, by overriding the equals() method if needed.
LSP :)